packages/core/src/behavior.ts

```diff
@@ -276,19 +286,23 @@ export function createMachineBehavior<TContext, TEvent extends EventObject>(
         }
         service.start();
       } else if (signal === stopSignal) {
-        service.stop();
+        service?.stop();
         subscription && subscription.unsubscribe(); // TODO: might not be necessary
       }
       return behavior;
     },
     receive: (_, event) => {
-      service.send(event);
+      service?.send(event);
       return behavior;
     },
     subscribe: (observer) => {
       return service?.subscribe(observer);
     },
-    current: machine.initialState // TODO: this should get from machine.getInitialState(ref)
+    get current() {
```

Comment:
shouldnt this be read from the created `service`? right now this won't ever be updated as machines are stateless

Reply:
This is badly named; it should only represent the `initial` state